src/main/java/algorithms/utilities/WordNet.java

```diff
@@ -34,6 +36,7 @@
 	
 	//Note that the WNDomains classification relies on WordNet-2.0
 	static RiWordNet database = new RiWordNet("./files/WordNet_2.0/dict");
+	static Map<String, Set<String>> synonymMemo = new HashMap<String, Set<String>>();
```

Comment:
Are we certain that this one is reset between every request? If not, it is imperative that it is changed. A static variable that is accessed staticly, and not manually reset, will not ever be reset as long as the java instance is running, and the java instance doesn't get shut down between requests, only the request instances.

Reply:
Its a win-win anyway. If it's reset between every request then atleast it helps caching for that reset. If it's not reset between every request thats even better since we are getting the synonyms from WordNet, therefore it would be the same result for every request.